src/kbmod/standardizers/standardizer.py

```diff
@@ -0,0 +1,589 @@
+        away, by avoiding acquiring or constructing the resource again during
+        initialization.
+
+        On a practical example: `FitsStandardizer` standardize FITS files
+        pointed to by a local filesystem path (i.e. the target). To confirm
+        their ability to standardize the target, they have to make sure they
+        can open and read the file. They use AstroPy to construct an
+        `~astropy.io.fits.HDUList`, i.e. the resource. Returning the already
+        constructed `~astropy.io.fits.HDUList` allows `FitsStandardizer` to
+        skip constructing a new one via `astropy.io.fits.open`.
+
+        Parameters
+        ----------
+        tgt : data-source
+            Some data source, URL, URI, POSIX filepath, Rubin Data Butler Data
+            Repository, an Python object, callable, etc.
+
+        Returns
+        -------
+        canStandardize : `bool`
+            `True` when the processor knows how to handle uploaded
+            file and `False` otherwise.
+        resources : `dict`, optional
+            Any optional resources constructed during target resolution. An
+            empty dictionary if there are none.
+        """
+        raise NotImplementedError()
+
+    @classmethod
+    def canStandardize(cls, tgt):
+        """Returns ``True`` when the standardizer knows how to process
+        the given upload (file) type.
+
+        Parameters
+        ----------
+        tgt : data-source
+            Some data source, URL, URI, POSIX filepath, Rubin Data Butler Data
+            Repository, an Python object, callable, etc.
+
+        Returns
+        -------
+        canStandardize : `bool`
+            `True` when the processor knows how to handle uploaded
+            file and `False` otherwise.
+
+        Notes
+        -----
+        Implementation is instrument-specific.
+        """
+        return cls.resolveTarget(tgt)[0]
+
+    def __init_subclass__(cls, **kwargs):
+        # Registers subclasses of this class with set `name` class
+        # parameters as availible standardizers. Note the specific
+        # standardizer has to be imported before this class since the
+        # mechanism is triggered at definition time.
+        name = getattr(cls, "name", False)
+        if name and name is not None:
+            super().__init_subclass__(**kwargs)
+            Standardizer.registry[cls.name] = cls
+
+    @abc.abstractmethod
+    def __init__(self, location, config=None, **kwargs):
+        self.location = location
+        self.processable = []
+        self.config = self.configClass(config)
+
+    def __str__(self):
+        return f"{self.name}({self.location}, {self.processable})"
+
+    def __repr__(self):
+        return f"{self.__class__.__name__}({self.location})"
+
+    # all these should probably be named in plural - but lord allmighty that's
+    # a lot of references to update and it feels so unnatural - help? These lead
```

Comment:
In cases like this, I usually name the function something like `wcs_list` instead of `wcses` (is that even how you'd spell the plural). I don't think it is worth updating all the references though.

Reply:
Yeah, I've no idea either :man_shrugging: Happy to update at a later point if needed.